src/Action.hs

```diff
@@ -251,9 +253,11 @@ currentUser = do
     uid <- currentUserId
     muser <- query (findUser uid)
     case muser of
-        Just user -> pure user
-        Nothing   -> logout >> throwError500 "Unknown user identitifer"
+        Just user | isActiveUser user
+            -> pure user
+        _   -> logout >> throwError500 "Unknown user identitifer"
 
+-- TODO: Check if the current user is not deleted.
```

Comment:
if the current user is deleted, something else has gone wrong!  (-:  So if you check it again here, it should be obvious that if the check fails it's an internal error.